src/main.c

```diff
@@ -128,14 +139,17 @@ void handleApdu(volatile unsigned int *flags, volatile unsigned int *tx, int rx)
             switch (e & 0xF000) {
                 case 0x6000:
                     sw = e;
+                    G_called_from_swap = false;
```

Comment:
Why copy paste `G_called_from_swap = false;` in all switch cases and not just do it before the switch  ?

Reply:
Also, why do you need to set this variable to `false` here  ?

Reply:
True

Reply:
I've moved the clear of the G_called_from_swap to the FINALLY()